alicloud/resource_alicloud_cen_bandwidth_package.go

```diff
@@ -0,0 +1,308 @@
+package alicloud
+
+import (
+	"fmt"
+	"strings"
+
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cbn"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAlicloudCenBandwidthPackage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCenBandwidthPackageCreate,
+		Read:   resourceAlicloudCenBandwidthPackageRead,
+		Update: resourceAlicloudCenBandwidthPackageUpdate,
+		Delete: resourceAlicloudCenBandwidthPackageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bandwidth": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(int)
+					if value < 1 {
+						errors = append(errors, fmt.Errorf("%s cannot be smaller than 1Mbps", k))
+					}
+
+					return
+				},
+			},
+
+			"geographic_region_ids": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				MaxItems: 2,
+				MinItems: 1,
+			},
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) < 2 || len(value) > 128 {
+						errors = append(errors, fmt.Errorf("%s cannot be shorter than 2 characters or longer than 128 characters", k))
+					}
+
+					if strings.HasPrefix(value, "http://") || strings.HasPrefix(value, "https://") {
+						errors = append(errors, fmt.Errorf("%s cannot starts with http:// or https://", k))
+					}
+
+					return
+				},
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if len(value) < 2 || len(value) > 256 {
+						errors = append(errors, fmt.Errorf("%s cannot be shorter than 2 characters or longer than 256 characters", k))
+					}
+
+					if strings.HasPrefix(value, "http://") || strings.HasPrefix(value, "https://") {
+						errors = append(errors, fmt.Errorf("%s cannot starts with http:// or https://", k))
+					}
+
+					return
+				},
+			},
+
+			"charge_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "PostPaid",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if value != "PrePaid" && value != "PostPaid" {
+						errors = append(errors, fmt.Errorf("%s must be one of: PrePaid or PostPaid", k))
+					}
+					return
+				},
+			},
+
+			"period": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
```

Comment:
Please add a DiffSuppressFunc and you can refer to: https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_instance.go#L154